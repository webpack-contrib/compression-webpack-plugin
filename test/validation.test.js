import CompressionPlugin from '../src';

it('validation', () => {
  /* eslint-disable no-new */
  expect(() => {
    new CompressionPlugin({ test: /foo/ });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ test: 'foo' });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ test: [/foo/, /bar/] });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ test: ['foo', 'bar'] });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ test: [/foo/, 'bar'] });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ test: true });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CompressionPlugin({ test: [true] });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CompressionPlugin({ test: [/foo/, 'foo', true] });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CompressionPlugin({ include: 'foo' });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ include: [/foo/, /bar/] });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ include: ['foo', 'bar'] });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ include: [/foo/, 'bar'] });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ include: true });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CompressionPlugin({ include: [true] });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CompressionPlugin({ include: [/foo/, 'foo', true] });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CompressionPlugin({ exclude: 'foo' });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ exclude: [/foo/, /bar/] });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ exclude: ['foo', 'bar'] });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ exclude: [/foo/, 'bar'] });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ exclude: true });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CompressionPlugin({ exclude: [true] });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CompressionPlugin({ exclude: [/foo/, 'foo', true] });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CompressionPlugin({ cache: true });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ cache: '/path/to/cache' });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ cache: () => {} });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CompressionPlugin({ filename: '[path].gz[query]' });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({
      filename() {
        return 'test';
      },
    });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ filename: true });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CompressionPlugin({ algorithm: 'gzip' });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({
      algorithm(input, compressionOptions, callback) {
        return callback(input);
      },
    });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ algorithm: true });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CompressionPlugin({ compressionOptions: { level: 1 } });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ compressionOptions: { unknown: 1 } });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ compressionOptions: '1024' });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CompressionPlugin({ threshold: 1024 });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ threshold: '1024' });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CompressionPlugin({ minRatio: 0.8 });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ minRatio: '0.8' });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CompressionPlugin({ deleteOriginalAssets: true });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ deleteOriginalAssets: false });
  }).not.toThrow();

  expect(() => {
    new CompressionPlugin({ deleteOriginalAssets: 'true' });
  }).toThrowErrorMatchingSnapshot();
  /* eslint-enable no-new */
});
