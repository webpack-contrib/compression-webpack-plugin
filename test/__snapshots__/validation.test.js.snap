// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation 1`] = `
"Compression Plugin Invalid Options

options.test should pass \\"instanceof\\" keyword validation
options.test should be string
options.test should match some schema in anyOf
options.test should be array
options.test should match some schema in anyOf
"
`;

exports[`validation 2`] = `
"Compression Plugin Invalid Options

options.test should pass \\"instanceof\\" keyword validation
options.test should be string
options.test should match some schema in anyOf
options.test.0 should pass \\"instanceof\\" keyword validation
options.test.0 should be string
options.test.0 should match some schema in anyOf
options.test.0 should match some schema in anyOf
options.test should match some schema in anyOf
"
`;

exports[`validation 3`] = `
"Compression Plugin Invalid Options

options.test should pass \\"instanceof\\" keyword validation
options.test should be string
options.test should match some schema in anyOf
options.test.2 should pass \\"instanceof\\" keyword validation
options.test.2 should be string
options.test.2 should match some schema in anyOf
options.test.2 should match some schema in anyOf
options.test should match some schema in anyOf
"
`;

exports[`validation 4`] = `
"Compression Plugin Invalid Options

options.include should pass \\"instanceof\\" keyword validation
options.include should be string
options.include should match some schema in anyOf
options.include should be array
options.include should match some schema in anyOf
"
`;

exports[`validation 5`] = `
"Compression Plugin Invalid Options

options.include should pass \\"instanceof\\" keyword validation
options.include should be string
options.include should match some schema in anyOf
options.include.0 should pass \\"instanceof\\" keyword validation
options.include.0 should be string
options.include.0 should match some schema in anyOf
options.include.0 should match some schema in anyOf
options.include should match some schema in anyOf
"
`;

exports[`validation 6`] = `
"Compression Plugin Invalid Options

options.include should pass \\"instanceof\\" keyword validation
options.include should be string
options.include should match some schema in anyOf
options.include.2 should pass \\"instanceof\\" keyword validation
options.include.2 should be string
options.include.2 should match some schema in anyOf
options.include.2 should match some schema in anyOf
options.include should match some schema in anyOf
"
`;

exports[`validation 7`] = `
"Compression Plugin Invalid Options

options.exclude should pass \\"instanceof\\" keyword validation
options.exclude should be string
options.exclude should match some schema in anyOf
options.exclude should be array
options.exclude should match some schema in anyOf
"
`;

exports[`validation 8`] = `
"Compression Plugin Invalid Options

options.exclude should pass \\"instanceof\\" keyword validation
options.exclude should be string
options.exclude should match some schema in anyOf
options.exclude.0 should pass \\"instanceof\\" keyword validation
options.exclude.0 should be string
options.exclude.0 should match some schema in anyOf
options.exclude.0 should match some schema in anyOf
options.exclude should match some schema in anyOf
"
`;

exports[`validation 9`] = `
"Compression Plugin Invalid Options

options.exclude should pass \\"instanceof\\" keyword validation
options.exclude should be string
options.exclude should match some schema in anyOf
options.exclude.2 should pass \\"instanceof\\" keyword validation
options.exclude.2 should be string
options.exclude.2 should match some schema in anyOf
options.exclude.2 should match some schema in anyOf
options.exclude should match some schema in anyOf
"
`;

exports[`validation 10`] = `
"Compression Plugin Invalid Options

options.cache should be boolean
options.cache should be string
options.cache should match some schema in anyOf
"
`;

exports[`validation 11`] = `
"Compression Plugin Invalid Options

options.filename should be string
options.filename should pass \\"instanceof\\" keyword validation
options.filename should match some schema in anyOf
"
`;

exports[`validation 12`] = `
"Compression Plugin Invalid Options

options.algorithm should be string
options.algorithm should pass \\"instanceof\\" keyword validation
options.algorithm should match some schema in anyOf
"
`;

exports[`validation 13`] = `
"Compression Plugin Invalid Options

options.compressionOptions should be object
"
`;

exports[`validation 14`] = `
"Compression Plugin Invalid Options

options.threshold should be number
"
`;

exports[`validation 15`] = `
"Compression Plugin Invalid Options

options.minRatio should be number
"
`;

exports[`validation 16`] = `
"Compression Plugin Invalid Options

options.deleteOriginalAssets should be boolean
"
`;

exports[`validation 17`] = `
"Compression Plugin Invalid Options

options should NOT have additional properties
"
`;
